rtl/src/main/scala/tensil/blackbox/xilinx/RAMMacro.scala

```diff
@@ -0,0 +1,155 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/* Copyright © 2019-2022 Tensil AI Company */
+
+package tensil.blackbox.xilinx
+
+import chisel3._
+import chisel3.util.HasBlackBoxInline
+import tensil.util
+import tensil.ArtifactsLogger
+
+object RAMMacro {
+  val UltraPrimitive = "ultra"
+  val BlockPrimitive = "block"
+}
+
+class RAMMacro(width: Int, depth: Long, primitive: String)
```

Comment:
Can you add a short scaladoc here capturing the info that `primitive` needs to be one of the strings in the `RAMMacro` companion object? I know that's obvious from the source but I'm certain I'll run into this at some point in the future and not notice